### PR TITLE
Update: only render modal if it should show

### PR DESCRIPTION
**Pull request by jrios**
When there are multiple modals on a page, it renders each one on the page. After a certain point, performance degrades and pages become less responsive.

Only rendering if the show prop is true will prevent every modal from being rendered and only render one modal on the page at any given time.

### CODE REVIEW

File: packages/es-components/src/components/containers/modal/Modal.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ function Modal(props) {
 
   const shouldCloseOnOverlayClick = backdrop !== 'static' && backdrop;
 
-  return (
+  return show ? (
```
Comment (Darrken) [inline packages/es-components/src/components/containers/modal/Modal.js:123]: I prefer the `show && (` syntax that doesn't require the `: null` at the end.

Comment (jrios) [reply packages/es-components/src/components/containers/modal/Modal.js:123]: 🤷‍♂️ I like the explicitness.

**Comment by jrios** [comment]
I did check. everything looked good from what I could tell.

File: packages/es-components/src/components/containers/modal/Modal.js (hunk 1 above)
Comment (Darrken) [reply packages/es-components/src/components/containers/modal/Modal.js:123]: Super.

**Pull request merged**